scanpy/external/pp/_scrublet.py

```diff
@@ -357,8 +361,22 @@ def _scrublet_call_doublets(
         random_state=random_state,
     )
 
-    scrub._E_obs_norm = adata_obs.X
-    scrub._E_sim_norm = adata_sim.X
+    # Ensure normalised matrix sparseness as Scrublet does
+    # https://github.com/swolock/scrublet/blob/67f8ecbad14e8e1aa9c89b43dac6638cebe38640/src/scrublet/scrublet.py#L100
+
+    def sparsify(mat):
+        if not sp.sparse.issparse(mat):
+            return sp.sparse.csc_matrix(mat)
+
+        elif not sp.sparse.isspmatrix_csc(mat):
+            return mat.tocsc()
+
+        else:
+            return mat
+
+    scrub._E_obs_norm = sparsify(adata_obs.X)
+    scrub._E_sim_norm = sparsify(adata_sim.X)
```

Comment:
```suggestion
    scrub._E_obs_norm = sp.sparse.csc_matrix(adata_obs.X)
    scrub._E_sim_norm = sp.sparse.csc_matrix(adata_sim.X)
```

The `csc_matrix` constructor doesn't copy by default, you'd have to pass `copy=True` for that:

```python
In [3]: a = sparse.random(100, 50, format="csc")

In [4]: b = sparse.csc_matrix(a)

In [5]: a
Out[5]: 
<100x50 sparse matrix of type '<class 'numpy.float64'>'
	with 50 stored elements in Compressed Sparse Column format>

In [6]: a.data[0]
Out[6]: 0.18339411768371083

In [7]: b.data[0] = 1

In [8]: a.data[0]
Out[8]: 1.0
```

Reply:
Thanks @ivirshup - was just copying (poorly) from Scrublet.